crates/compilers/src/zksync/artifact_output/zk/bytecode.rs

```diff
@@ -30,7 +30,7 @@ impl ZkArtifactBytecode {
         Self { object, is_unlinked, missing_libraries }
     }
 
-    fn link_references(&self) -> BTreeMap<String, BTreeMap<String, Vec<Offsets>>> {
+    pub fn link_references(&self) -> BTreeMap<String, BTreeMap<String, Vec<Offsets>>> {
```

Comment:
Could you explain a little bit why link references are still used with *zksolc*?

Reply:
Well, we do not make use of it in the same was as `solc`, but rather that's how we signal `forge` that the contract is unlinked and which libraries are needed.
In the future (see https://github.com/matter-labs/foundry-zksync/issues/723) this information could be used to pass the needed libraries to the `zksolc` invocation for linking (which is not here yet either)

Reply:
The `offsets` marks where in the bytecode the library is referenced so when solc-style linking happens foundry doesn't need to search _again_ where the library is referenced and can just replace those sections of the bytecode

Reply:
So yeah we put an empty list of offsets here, as a list is expected regardless